### PR TITLE
CFD state machine as of version `0.1.1`

### PR DESCRIPTION
**Pull request by da-kami**
Note that this state machine should be evolved during the refactoring of the model, so we keep track of the changes.

### CODE REVIEW

**Review by DeliciousHair — APPROVED** [review]
LGTM!

File: docs/asset/mvp_cfd_state_machine.puml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+Open --> OpenCommitted
+PendingCommit --> OpenCommitted
+PendingOpen --> OpenCommitted
+OpenCommitted --> OpenCommitted
+
+Open --> MustRefund
+OpenCommitted --> MustRefund
+PendingOpen --> MustRefund
+
+MustRefund --> Refunded
```
Comment (klochowicz) [inline docs/asset/mvp_cfd_state_machine.puml]:
```suggestion
PendingRefund --> Refunded
```

File: docs/asset/mvp_cfd_state_machine.puml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ContractSetup --> PendingOpen
+
+PendingCommit --> PendingCommit
+
+Open --> OpenCommitted
+PendingCommit --> OpenCommitted
+PendingOpen --> OpenCommitted
+OpenCommitted --> OpenCommitted
+
+Open --> MustRefund
```
Comment (klochowicz) [inline docs/asset/mvp_cfd_state_machine.puml]:
```suggestion
Open --> PendingRefund
```

File: docs/asset/mvp_cfd_state_machine.puml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+OpenCommitted --> MustRefund
```
Comment (klochowicz) [inline docs/asset/mvp_cfd_state_machine.puml]:
```suggestion
OpenCommitted --> PendingRefund
```

File: docs/asset/mvp_cfd_state_machine.puml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+PendingOpen --> MustRefund
```
Comment (klochowicz) [inline docs/asset/mvp_cfd_state_machine.puml]:
```suggestion
PendingOpen --> PendingRefund
```

File: docs/asset/mvp_cfd_state_machine.puml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+MustRefund --> Refunded
+
+PendingOpen --> PendingCET
+Open --> PendingCET
+PendingCommit --> PendingCET
+OpenCommitted --> PendingCET
+PendingCET --> PendingCET
+MustRefund --> PendingCET
```
Comment (klochowicz) [inline docs/asset/mvp_cfd_state_machine.puml]:
```suggestion
PendingRefund --> PendingCET
```

File: docs/asset/mvp_cfd_state_machine.puml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+Open --> PendingCommit
+PendingOpen --> PendingCommit
+
+PendingOpen --> Closed
+Open --> Closed
+PendingCommit --> Closed
+OpenCommitted --> Closed
+PendingCET --> Closed
+MustRefund --> Closed
```
Comment (klochowicz) [inline docs/asset/mvp_cfd_state_machine.puml]:
```suggestion
PendingRefund --> Closed
```

**Review by klochowicz — APPROVED** [review]
looks good, I've just borsed a small refactor, so just raised suggestions to keep it aligned :)

**Comment by da-kami** [comment]
Ok, merging it! In case we notice that something was not depicted as the code behaves we can always open another PR.

**Comment by da-kami** [comment]
I decided to add a bit more information to make things a bit clearer.

**Comment by da-kami** [comment]
Sorry for the complexity, but without that it would have been impossible to validate it 😅

**Comment by DeliciousHair** [comment]
> Sorry for the complexity, but without that it would have been impossible to validate it sweat_smile

Complex?!? All the information is present and clear, I'd say this reduces complexity immensely!

**Review by DeliciousHair — APPROVED** [review]
LGTM!

**Pull request merged**